Clarify naming conventions for link types

### Description
I couldn't find any explicit naming conventions in regards to link types in the [event guidelines](https://github.com/eiffel-community/eiffel/blob/master/eiffel-syntax-and-usage/the-links-object.md).

All existing link types seems to be nouns, is this some unwritten rule? Are verbs not allowed? I'm missing documentation about what is allowed and what is not.

### Motivation
It will be easier to understand when/if someone wants extend the protocol.

### Exemplification
Extend the documentation about the links object with a clear rule saying what is allowed or not.

### Benefits
Clarified documentation.

### Possible Drawbacks
None that I can think of.

Below is an extraction of all link types for all (source) events and I can't find one that doesn't fit the previously described pattern.

```
$ for i in *.md ; do echo $i | sed 's/\.md//' ; sed -n 's/^### \([A-Z][A-Z_]*$\)/    \1/p' $i ; done
EiffelActivityCanceledEvent
    ACTIVITY_EXECUTION
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelActivityFinishedEvent
    ACTIVITY_EXECUTION
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelActivityStartedEvent
    ACTIVITY_EXECUTION
    PREVIOUS_ACTIVITY_EXECUTION
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelActivityTriggeredEvent
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelAnnouncementPublishedEvent
    MODIFIED_ANNOUNCEMENT
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelArtifactCreatedEvent
    COMPOSITION
    ENVIRONMENT
    PREVIOUS_VERSION
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelArtifactPublishedEvent
    ARTIFACT
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelArtifactReusedEvent
    COMPOSITION
    REUSED_ARTIFACT
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelCompositionDefinedEvent
    ELEMENT
    PREVIOUS_VERSION
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelConfidenceLevelModifiedEvent
    SUBJECT
    SUB_CONFIDENCE_LEVEL
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelEnvironmentDefinedEvent
    PREVIOUS_VERSION
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelFlowContextDefinedEvent
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelIssueDefinedEvent
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelIssueVerifiedEvent
    SUCCESSFUL_ISSUE
    FAILED_ISSUE
    INCONCLUSIVE_ISSUE
    IUT
    VERIFICATION_BASIS
    ENVIRONMENT
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelSourceChangeCreatedEvent
    BASE
    PREVIOUS_VERSION
    PARTIALLY_RESOLVED_ISSUE
    RESOLVED_ISSUE
    DERESOLVED_ISSUE
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelSourceChangeSubmittedEvent
    CHANGE
    PREVIOUS_VERSION
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelTestCaseCanceledEvent
    TEST_CASE_EXECUTION
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelTestCaseFinishedEvent
    TEST_CASE_EXECUTION
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelTestCaseStartedEvent
    TEST_CASE_EXECUTION
    ENVIRONMENT
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelTestCaseTriggeredEvent
    IUT
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelTestExecutionRecipeCollectionCreatedEvent
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelTestSuiteFinishedEvent
    TEST_SUITE_EXECUTION
    CAUSE
    CONTEXT
    FLOW_CONTEXT
EiffelTestSuiteStartedEvent
    TERC
    CAUSE
    CONTEXT
    FLOW_CONTEXT
```

--- a/eiffel-syntax-and-usage/event-design-guidelines.md
+++ b/eiffel-syntax-and-usage/event-design-guidelines.md
@@ -33,7 +33,4 @@ The design of Eiffel events is governed by the following guidelines:
 * __Information model integrity over reader convenience:__ Eiffel events are not convenient to read – or to publish – as they require graph traversal in order to derive any meaningful information. Do not work around this by including convenience information (typically duplication). Such measures will never be completely satisfactory. Instead the primary concern of event design is information model integrity; readability and usability is the concern of services built upon that data source.
 * __Key names in camelCase:__ All key names shall use camelCase.
 * __Enumerations in CAPS_WITH_UNDERSCORE:__ All enumerated values shall use CAPS_WITH_UNDERSCORE.
-
-
-
-
+* __Link types are nouns:__ Link types shall have names that fit the pattern "\<target event> is the \<source event>'s \<link type>", implying that they are nouns that describe the relationship between the source and the target event.

